core/model/modx/modcachemanager.class.php

```diff
@@ -756,4 +756,22 @@ public function clearCache(array $paths= array(), array $options= array()) {
 
         return $results;
     }
+
+    /**
+     * Flush permissions for users
+     *
+     * @return string|array The success response
+     */
+    public function flushPermissions() {
+        $ctxQuery = $this->modx->newQuery('modContext');
+        $ctxQuery->select($this->modx->getSelectColumns('modContext', '', '', array('key')));
+        if ($ctxQuery->prepare() && $ctxQuery->stmt->execute()) {
+            $contexts = $ctxQuery->stmt->fetchAll(PDO::FETCH_COLUMN);
+            if ($contexts) {
+                $serialized = serialize($contexts);
+                $this->modx->exec("UPDATE {$this->modx->getTableName('modUser')} SET {$this->modx->escape('session_stale')} = {$this->modx->quote($serialized)}");
+            }
+        }
+        return $this->modx->error->success('');
```

Comment:
This would be "useless" since the "exec" never sets an error.

What do you think about returning `false` as default (meaning we were unable to perform the statement, and return `true` whenever `modX->exec` is executed ?